src/main/java/de/tum/in/test/api/ast/type/ExceptionHandlingType.java

```diff
@@ -0,0 +1,54 @@
+package de.tum.in.test.api.ast.type;
+
+import com.github.javaparser.ast.Node;
+import com.github.javaparser.ast.stmt.*;
+import com.github.javaparser.utils.Pair;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public enum ExceptionHandlingType {
+    /**
+     * Returns a list with all exception handling types
+     */
+    ALL(new ArrayList<>() {{
+        add(new Pair<>("Assert-Statement", AssertStmt.class));
+        add(new Pair<>("Throw-Statement", ThrowStmt.class));
+        add(new Pair<>("Try-Statement", TryStmt.class));
+        add(new Pair<>("Catch-Statement", CatchClause.class));
+    }}),
+    /**
+     * Returns the assert type
+     */
+    ASSERT(new ArrayList<>() {{
+        add(new Pair<>("Assert-Statement", AssertStmt.class));
+    }}),
+    /**
+     * Returns the throw type
+     */
+    THROW(new ArrayList<>() {{
+        add(new Pair<>("Throw-Statement", ThrowStmt.class));
+    }}),
+    /**
+     * Returns the try-catch exception handling type
+     */
+    TRYCATCH(new ArrayList<>() {{
+        add(new Pair<>("Try-Statement", TryStmt.class));
+        add(new Pair<>("Catch-Statement", CatchClause.class));
+    }});
```

Comment:
This is a bit more subtle, as we need to discuss if `try { ... } finally { ... }` is really exception handling. To some degree it is, but it does neither catch, nor throw, nor delegate exception more than regular code.

What would be the use case / what do you want it to do?

Reply:
I intended to keep the amount of `Type` implementing enums smaller in order to avoid loosing clarity. Therefore `try { ... } catch { ... } ` fits rather in the exception handling category than in the class, conditional or loop category.

A use case would be `ClassWithAnyKindsOfExceptionHandlings.java`, where we would like to check whether the method `tryCatchStatement()`contains the unwanted nodes `try` and `catch`. This could be useful in case we have an exercise where we want students to just throw the exception further rather than handling it.